src/mbed_tools/targets/get_target.py

```diff
@@ -26,15 +25,16 @@ def get_target_by_name(name: str, path_to_targets_json: pathlib.Path) -> Target:
     Raises:
         TargetError: an error has occurred while fetching target
     """
-    return Target.from_targets_json(name, path_to_targets_json)
+    try:
+        return target_attributes.get_target_attributes(path_to_targets_json, name)
+    except (FileNotFoundError, target_attributes.TargetAttributesError) as e:
+        raise TargetError(e) from e
```

Comment:
Should we add a test to pretend to cause a file not found and check we get TargetError?

Reply:
Yeah, this is also why codecov is failing and why the PR is still marked "WIP".

Reply:
Added tests to check `TargetError` is raised when we hit `FileNotFound` or `TargetAttributesError`.